### ISSUE TITLE
Build v2.1.0.4 fails: "none" is not a known value of feature <optimization>

### ISSUE THREAD

**Issue by NoAgendaIT** [Gulden/gulden-official#164]
### Describe the issue

This is a Funtoo Linux machine, x64, with gcc 7.4.1, glibc 2.27.

I have pulled gulden from the github repository, checked out tag v2.1.0.4, and I have followed the build instructions and checked the dependencies that are appropriate for this distribution.

I ran `./autogen`, followed by `cd depends`, followed by `make`.

The compilation fails with the following error:

```
(...)
Building boost...
/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/boost/1_66_0-d56125989ed/tools/build/src/build/feature.jam:494: in feature.validate-value-string from module feature
error: "none" is not a known value of feature <optimization>
error: legal values: "off" "speed" "space"
/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/boost/1_66_0-d56125989ed/tools/build/src/build/property.jam:276: in validate1 from module property
/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/boost/1_66_0-d56125989ed/tools/build/src/build/property.jam:302: in property.validate from module property
/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/boost/1_66_0-d56125989ed/tools/build/src/tools/features/variant-feature.jam:70: in variant from module variant-feature
/usr/share/boost-build/site-config.jam:9: in modules.load from module site-config
/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/boost/1_66_0-d56125989ed/tools/build/src/build-system.jam:249: in load-config from module build-system
/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/boost/1_66_0-d56125989ed/tools/build/src/build-system.jam:351: in load-configuration-files from module build-system
/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/boost/1_66_0-d56125989ed/tools/build/src/build-system.jam:524: in load from module build-system
/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/boost/1_66_0-d56125989ed/tools/build/src/kernel/modules.jam:295: in import from module modules
/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/boost/1_66_0-d56125989ed/tools/build/src/kernel/bootstrap.jam:139: in boost-build from module
/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/boost/1_66_0-d56125989ed/boost-build.jam:17: in module scope from module

make: *** [funcs.mk:244: /src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/boost/1_66_0-d56125989ed/./.stamp_built] Error 1
```

**Comment by mjmacleod** [comment]
This appears to be due to funtoo/gentoo defining a "/usr/share/boost-build/site-config.jam" - that is interfering with the build process.

Try adding "--ignore-site-config" after the 'b2' calls in 'depend/packages/boost.mk'

**Comment by NoAgendaIT** [comment]
Ah, indeed. Thanks. This allows the build to continue. Perhaps a good idea to add this to the docs? :grin:

**Comment by mjmacleod** [comment]
I've pushed it into the development branch now, so should be included in next release (there will be a 2.1.0.5 in the next few days)

In the unlikely event that this causes issues elsewhere and needs to be reverted, I  will definitely document it instead at that point, either way it should be fixed.

Thanks for the feedback.
